### PR TITLE
Fix versionlock plugin detection for yum/dnf - RHEL/CentOS 8

### LINKED ISSUE #54798: Centos 8 pkg hold error yum-plugin-versionlock not installed.
### Description of Issue
On a fresh install Centos 8 with installing packages with `hold: false` results in an error:
[ERROR   ] Cannot proceed, yum-plugin-versionlock is not installed.

My initial thoughts, is that salt is using yum on centos-8 but it should use dnf.

### Setup
packages-formulas
```
54  wanted_pkgs:
1   pkg.installed:
2     - pkgs: {{ wanted_packages | json }}
3     {% if grains['os_family'] not in ['Suse'] %}
4     - hold: false
5     {% endif %}
6     - require:
7       - pkg: pkg_req_pkgs
8       {% if req_states %}
9         {% for dep in req_states %}
10       - sls: {{ dep }}
11         {% endfor %}
12       {% endif %}
13     - retry: {{ packages.retry_options|json }}
```

### Steps to Reproduce Issue
install salt-minion on centos8 and use packages-formula to install list of packages.

### Versions Report

Salt Version:
Salt: 2019.2.1
Dependency Versions:
cffi: 1.11.5
cherrypy: Not Installed
dateutil: 2.6.1
docker-py: Not Installed
gitdb: Not Installed
gitpython: Not Installed
ioflo: Not Installed
Jinja2: 2.10.1
libgit2: Not Installed
libnacl: Not Installed
M2Crypto: 0.33.0
Mako: Not Installed
msgpack-pure: Not Installed
msgpack-python: 0.6.1
mysql-python: Not Installed
pycparser: 2.14
pycrypto: Not Installed
pycryptodome: Not Installed
pygit2: Not Installed
Python: 3.6.8 (default, Jul  1 2019, 16:43:04)
python-gnupg: Not Installed
PyYAML: 3.12
PyZMQ: 17.0.0
RAET: Not Installed
smmap: Not Installed
timelib: Not Installed
Tornado: 4.5.2
ZMQ: 4.3.1
System Versions:
dist: centos 8.0.1905 Core
locale: UTF-8
machine: x86_64
release: 4.18.0-80.7.1.el8_0.x86_64
system: Linux
version: CentOS Linux 8.0.1905 Core

### PR DESCRIPTION
**Pull request by burningalchemist**
### What does this PR do?
Adds versionlock plugin detection for RHEL8/CentOS 8

### What issues does this PR fix or reference?
fixes #54798

### Tests written?
Yes

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by burningalchemist** [comment]
Fixing linting...

**Comment by burningalchemist** [comment]
Hi @frogunder , I think it's ready for a review.
I can squash fixups later and update the branch to run another round of tests, if it's ok.

File: salt/modules/yumpkg.py [hunk 1]
```diff
@@ -145,7 +145,9 @@ def _yum():
     contextkey = 'yum_bin'
     if contextkey not in __context__:
         if ('fedora' in __grains__['os'].lower()
-           and int(__grains__['osrelease']) >= 22):
+           and int(__grains__['osrelease']) >= 22
+         ) or (__grains__['os'].lower() in ('redhat', 'centos')
```
Comment (burningalchemist) [inline salt/modules/yumpkg.py]: Somehow this bad formatting was missed. Will be fixed.

**Review by frogunder — DISMISSED** [review]

**Comment by burningalchemist** [comment]
@frogunder I've updated the PR to match Black code style. :)

**Review by dwoz — APPROVED** [review]

**Pull request merged**